### PR TITLE
Frontend associations pages

### PR DESCRIPTION
**Pull request by abocquet**

### CODE REVIEW

**Comment by Piwakk** [comment]
I think the folder `pages` should be renamed `components`
Also, let's capitalize the file name of `*.tsx` files, it looks like a (unofficial) convention, and the default `App.tsx` follows it. See Airbnb https://github.com/airbnb/javascript/tree/master/react#naming

File: frontend/src/models/user.ts [hunk 1]
```diff
@@ -5,6 +5,23 @@ export enum StudentType {
     Ic = "IC"
 }
 
+export interface Role {
+    id: number;
+    association: string;
+    role: string;
+    rank: number;
+
+    is_archived: boolean;
```
Comment (Piwakk) [inline frontend/src/models/user.ts]: isArchived

File: frontend/src/pages/associations/list.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    if (isLoading) return "Loading...";
+    if (error) {
+        return `Something went wrong: ${error.message}`;
+    }
+    if (associations) {
+        return (
+            <>
+                <PageTitle>Associations</PageTitle>
+                <div className={"row"}>
```
Comment (Piwakk) [inline frontend/src/pages/associations/list.tsx]: There is a component `Row`.
Also, [Bootstrap](https://getbootstrap.com/docs/4.3/layout/grid/) always puts it in a `Container` component (I think this is not the case here)

Comment (Piwakk) [reply frontend/src/pages/associations/list.tsx]: `Row` is in react-bootstrap at list, I don't know about reactstrap

File: frontend/src/pages/associations/main.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
+import React from "react";
```
Comment (Piwakk) [inline frontend/src/pages/associations/main.tsx]: We should either capitalize our component filenames, or switch `App.tsx` to `app.tsx`.
I think the capitalization is much more common, even with functional components.
See also https://github.com/airbnb/javascript/tree/master/react#naming

File: frontend/src/pages/associations/main.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            />
+        )
+    );
+
+    // Render
+    if (isLoading) return "Loading association container...";
+    if (error) return `Something went wrong: ${error.message}`;
+    if (association) {
+        return (
+            <div className={"container"}>
```
Comment (Piwakk) [inline frontend/src/pages/associations/main.tsx]: Use `Container` and `Row`, etc.

File: frontend/src/pages/associations/page/edit.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
+import React, { useContext } from "react";
+import { Page, PageType } from "../../../models/associations/page";
+import { useFormik } from "formik";
+import { Form, Button } from "react-bootstrap";
```
Comment (Piwakk) [inline frontend/src/pages/associations/page/edit.tsx]: `react-bootstrap` mentions another practice in the docs: https://react-bootstrap.github.io/getting-started/introduction/#importing

File: frontend/src/pages/associations/sidebar.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        <SideBarItem
+            icon={"plus"}
+            to={`/associations/${association.id}/pages/new`}
+        >
+            Ajouter une page
+        </SideBarItem>
+    );
+};
+
+export const AssociationSidebar = ({ association }) => {
```
Comment (Piwakk) [inline frontend/src/pages/associations/sidebar.tsx]: `Sidebar` or `SideBar`?
Actually, _sidebar_ is [a word](https://www.wordreference.com/enfr/sidebar), please change

File: frontend/src/pages/homepage.tsx [hunk 7]
```diff
@@ -1,14 +1,6 @@
...
 export const Homepage = () => {
-    return (
-        <>
-            <Navbar />
-            <div className={"container"}>
-                <PageTitle>Homepage</PageTitle>
-            </div>
-        </>
-    );
+    return <PageTitle>Homepage</PageTitle>;
```
Comment (Piwakk) [inline frontend/src/pages/homepage.tsx]: Is that all?
What is this component for?

File: frontend/src/routing/associations.ts [hunk 8]
```diff
@@ -0,0 +1,30 @@
+import { AssociationHome } from "../pages/associations/home";
+import { AssociationShowPage } from "../pages/associations/page/show";
+import { AssociationCreatePage, AssociationEditPage } from "../pages/associations/page/edit";
+
+export const routes = association => [
+    {
+        path: `/`,
+        component: AssociationHome,
+        exact: true,
+        props: { association: association, myValue: 3 }
```
Comment (Piwakk) [inline frontend/src/routing/associations.ts]: myValue?

File: frontend/src/routing/associations.ts [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+        exact: true,
+        props: { association: association }
+    },
+    {
+        path: `/pages/:pageId/edit`,
+        component: AssociationEditPage,
+        exact: true,
+        props: { association: association }
+    }
+];
```
Comment (Piwakk) [inline frontend/src/routing/associations.ts]: _No newline at end of file_

File: frontend/src/services/apiService.ts [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+const baseApi = "http://localhost:8000/api/v1";
+
+export const apiService = applyConverters(
+    Axios.create({
+        withCredentials: true,
+        baseURL: baseApi
+    })
+);
+
+function unwrap<T>(promise) {
```
Comment (Piwakk) [inline frontend/src/services/apiService.ts]: Maybe add a comment here to help the people not at ease with promises

File: frontend/src/utils/route.tsx [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+    );
+};
+
+export const CommonPrivateRoute = ({ component: Component, ...rest }) => (
+    <PrivateRoute
+        {...rest}
+        component={props => (
+            <>
+                <Navbar />
+                <div className={"container"}>
```
Comment (Piwakk) [inline frontend/src/utils/route.tsx]: `Container`

File: frontend/src/utils/sidebar.tsx [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+            </span>
+            {children}
+        </Link>
+    );
+};
+
+export const SideBar = ({ title, children, ...rest }) => {
+    return (
+        <>
+            <h1 className="page-title mb-5 mt-5">{title}</h1>
```
Comment (Piwakk) [inline frontend/src/utils/sidebar.tsx]: Not sure this is the title of the page

File: frontend/src/pages/homepage.tsx (hunk 7 above)
Comment (abocquet) [reply frontend/src/pages/homepage.tsx]: It's a WIP, to be changed in another PR

File: frontend/src/utils/sidebar.tsx (hunk 12 above)
Comment (abocquet) [reply frontend/src/utils/sidebar.tsx]: I'm not sure what you mean

File: frontend/src/pages/associations/Sidebar.tsx [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+};
+
+export const AssociationSidebar = ({ association }) => {
+    const { data: pages, isLoading, error } = useQuery(
+        ["pages.list", { associationId: association.id }],
+        api.pages.list
+    );
+
+    if (isLoading) {
+        return <p>Loading</p>;
```
Comment (Piwakk) [inline frontend/src/pages/associations/Sidebar.tsx]: Use French here?

File: frontend/src/pages/associations/Sidebar.tsx [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    if (error) {
+        return <p>Error while loading</p>;
```
Comment (Piwakk) [inline frontend/src/pages/associations/Sidebar.tsx]: French

File: frontend/src/pages/associations/page/Edit.tsx [hunk 15]
```diff
@@ -0,0 +1,124 @@
...
+export const AssociationEditPage = ({ association, ...props }) => {
+    const { pageId } = useParams();
+
+    const { data: page, isLoading, error } = useQuery(
+        ["page.get", { pageId }],
+        api.pages.get
+    );
+
+    if (isLoading) return "Loading association...";
+    if (error) return `Something went wrong: ${error.message}`;
```
Comment (Piwakk) [inline frontend/src/pages/associations/page/Edit.tsx]: FR

File: frontend/src/pages/associations/Main.tsx [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+                component={component}
+                key={path}
+                routeProps={props}
+            />
+        )
+    );
+
+    // Render
+    if (isLoading) return "Loading association container...";
+    if (error) return `Something went wrong: ${error.message}`;
```
Comment (Piwakk) [inline frontend/src/pages/associations/Main.tsx]: Français

**Pull request merged**